scheduler.py

```diff
@@ -89,6 +91,85 @@ def string_schedule(self):
 				date = date + timedelta(days=1)
 		return string
 
+	def json_schedule(self):
+		request_json = []
+
+		# Loop through schedule to create the skeleton JSON dictionaries
+
+		# Date for the first loop
+		json_date = self.start_date
+		if json_date is not None:
+			json_date = json_date + timedelta(days=1)
+
+		for i, schedule in enumerate(self.schedule):
+			# Get calendar day
+			actual_day = i + self.start_day
+
+			# Create first JSON dictionary if it does not exist
+			if len(request_json) == 0:
+				request_json.append({ "start_date": str(json_date).strip(), "schedule": [] })
+
+			# If in a new week, create a new JSON dictionary for the new week
+			if (actual_day % Chef.days_in_week == Chef.days_in_week - 1):
+				if json_date is not None and (i + 1 != self.num_days):
+					json_date = json_date + timedelta(days=2)
+					request_json.append({ "start_date": str(json_date).strip(), "schedule": [] })
+
+			# Increment the date
+			if json_date is not None:
+				json_date = json_date + timedelta(days=1)
+
+		# Loop through schedule to add the schedule data dictionaries to the appropriate
+		# JSON dictionaries
+
+		# Date for the second loop
+		date = self.start_date
+		if date is not None:
+			date = date + timedelta(days=1)
+		jsonToUpdate = 0
+
+		for i, schedule in enumerate(self.schedule):
+			# Get calendar day
+			actual_day = i + self.start_day
+
+			cooking_day = DAYS_OF_THE_WEEK(actual_day % Chef.days_in_week).name
```

Comment:
ah--I see. this is the benefit of an Enum. You didn't like my _day_of_week() method? ;)
The only potential problem i see with relying on the name of the enum is that if adam changes his contract, so he expects "THU" instead of "THURS", now you've got to go through the code looking for every occurence of THURS and changing it. If you instead had a method that specifically translates a number to a string for the explicit purpose of creating the json, you decouple that dependency. Up to you.

Reply:
Let's merge this and think about how to better do this later